### PR TITLE
Added: support for documents streams, objects, objects in sub arrays;…

### PR DESCRIPTION
**Pull request by OnkelTem**
… added basic tests.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+root = true
```
Comment (MAXakaWIZARD) [inline .editorconfig:1]: This file should be added to gitignore on OS level

File: src/Parser.php [hunk 2]
```diff
@@ -17,17 +17,18 @@ class Parser
...
     /**
-     * @param string $filePath Source file path
+     * @param string|mixed $input
      * @param callback|callable $itemCallback Callback
      * @param bool $assoc Parse as associative arrays
      *
      * @throws \Exception
      */
-    public function parse($filePath, $itemCallback, $assoc = true)
+    public function parse($input, $itemCallback, $assoc = true)
```
Comment (MAXakaWIZARD) [inline src/Parser.php:26]: This is breaking change. Let's introduce new method called `parseFromString` instead

**Comment by MAXakaWIZARD** [comment]
Also, build failed

**Comment by OnkelTem** [comment]
> Also, build failed

I wrote about this here: 
 https://github.com/MAXakaWIZARD/JsonCollectionParser/issues/3#issuecomment-346880811

**please do not accept it until the upstream is updated as described. The tests WILL fail until then.**

File: .editorconfig (hunk 1 above)
Comment (OnkelTem) [reply .editorconfig:1]: Why is that? It defines **your** code style, not your OS code style.

File: src/Parser.php (hunk 2 above)
Comment (OnkelTem) [reply src/Parser.php:26]: What does it break? It supports both old and new inputs.

File: .editorconfig (hunk 1 above)
Comment (MAXakaWIZARD) [reply .editorconfig:1]: This library is platform-independent. We don't need this `.editorconfig` file for sure. Please exclude it on your side via global `.gitignore`

**Comment by MAXakaWIZARD** [comment]
Support for new input formats is available in v1.6.0: https://github.com/MAXakaWIZARD/JsonCollectionParser/blob/master/CHANGELOG.md
Thanks, @OnkelTem

**Pull request closed without merging**